DigitalLearningSolutions.Web.Tests/Controllers/Signposting/ViewResourceControllerTests.cs

```diff
@@ -0,0 +1,86 @@
+﻿namespace DigitalLearningSolutions.Web.Tests.Controllers.Signposting
+{
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Controllers.Signposting;
+    using DigitalLearningSolutions.Web.Tests.ControllerHelpers;
+    using FakeItEasy;
+    using FluentAssertions.AspNetCore.Mvc;
+    using NUnit.Framework;
+
+    public class ViewResourceControllerTests
+    {
+        private ViewResourceController controller = null!;
+
+        private ILearningResourceReferenceDataService resourceDataService = null!;
+        private ILearningHubAuthApiUrlService urlService = null!;
+        private IUserService userService = null!;
+
+        [SetUp]
+        public void SetUp()
+        {
+            urlService = A.Fake<ILearningHubAuthApiUrlService>();
+            userService = A.Fake<IUserService>();
+            resourceDataService = A.Fake<ILearningResourceReferenceDataService>();
+
+            controller = new ViewResourceController(
+                    userService,
+                    resourceDataService,
+                    urlService
+                ).WithDefaultContext()
+                .WithMockUser(true);
+        }
+
+        [Test]
+        public void ViewResourceController_Index_returns_redirect_to_login_result_if_user_linked()
+        {
+            // Given
+            var authId = 1;
+            var resourceUrl = "De/Humani/Corporis/Fabrica";
+
+            A.CallTo(() => userService.GetDelegateUserLearningHubAuthId(A<int>._)).Returns(authId);
+            A.CallTo(() => resourceDataService.GetLearningHubResourceLinkById(5)).Returns(resourceUrl);

```

Comment:
Do you need to be explicit on the value passed in i.e. `5` ?

Reply:
It ensures the value passed into the controller method isn't accidentally misplaced or overwritten.

Reply:
In the setup you can just got for `A<int>._`. But it doesn't matter either way so leave as is if you prefer.
If you want to assert the method was actually called with the correct number then you could assert with `MustHaveHappenedOnceExactly()`